feat: add resetCameraScreenSpace method with offsetRatio input, so actors fully fit the screen

aerogt3 opened the pull request:
add resetCameraScreenSpace method with offsetRatio input, essentially a resetCamera call that fills the entire screen space (where 90% offsetRatio means visible actors will fill 90% of the screen space)

#1285

### Context
#1285 

### Results
See image here:
https://github.com/Kitware/vtk-js/issues/1285#issuecomment-3510396233

### Changes
Duplicated the resetCamera method, with logic to ensure visible actors fill a percentage of the screen (default: 90%)

### PR and Code Checklist
- [x] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [x] Run `npm run reformat` to have correctly formatted code

### Testing
- [x] Tested environment:
  - master
  - ubuntu 2240
  - firefox 144.0.2 & chrome 142.0

finetjul commented on Sources/Rendering/Core/Renderer/index.js:
```diff
@@ -464,6 +464,149 @@ function vtkRenderer(publicAPI, model) {
+
+    return true;
+  };
+
```
why not recoding the [VTK C++ implementation](https://gitlab.kitware.com/vtk/vtk/-/blob/master/Rendering/Core/vtkRenderer.cxx?ref_type=heads#L1306) ? 

This ensures you have the same behavior between VTK C++ and VTK.js. Which is important.

finetjul commented on Sources/Rendering/Core/Renderer/index.js:
```diff
@@ -464,144 +464,108 @@ function vtkRenderer(publicAPI, model) {
-      -center[1],
-      -center[2]
-    );
+    // ZoomToBoxUsingViewAngle
```
to move in its own method

finetjul reviewed (changes requested):
Thanks ! That's much easier to maintain.

skoloCFD committed:
feat: add resetCameraScreenSpace method with offsetRatio input

Port of VTK C++ vtkRenderer::ResetCameraScreenSpace. Uses a screen-space
bounding box to zoom closer to the data, correctly accounting for viewport
aspect ratio for both perspective and parallel projection.

Also adds zoomToBoxUsingViewAngle as a standalone public method.

Closes #1285

skoloCFD committed:
feat: add resetCameraScreenSpace method with offsetRatio input
Port of VTK C++ vtkRenderer::ResetCameraScreenSpace. Uses a screen-space
bounding box to zoom closer to the data, correctly accounting for viewport
aspect ratio for both perspective and parallel projection.

Also adds zoomToBoxUsingViewAngle as a standalone public method.

Closes #1285

aerogt3 commented on Sources/Rendering/Core/Renderer/index.js:
```diff
@@ -464,144 +464,108 @@ function vtkRenderer(publicAPI, model) {
+          ymax = Math.max(dy, ymax);
+        }
+      }
     }
```
The latest changes (including the order of the functions) should better match VTK C++
